esmvalcore/preprocessor/_ancillary_vars.py

```diff
@@ -46,6 +49,17 @@ def _load_fx(fx_info, check_level):
     return fx_cube
 
 
+def _is_fx_broadcastable(fx_cube, cube):
+    try:
+        da.broadcast_to(fx_cube.core_data(), cube.shape)
```

Comment:
am a bit confused as to why the broadcasting test, since all we need is to check if the fx data is time dependent or not: if it's not then a simple (X, Y) shapes check is enough, if it's time-dependent then we have to be very careful time points match and broadcasting might work but time points might differ so things get messy

Reply:
I think it's more straightforward to do that. It could be that the shapes do not match because the fx var and the var are on differents grids. So this is a way to take into account all these cases with just one test.

Reply:
OK but I am still concerned with two cases: 1. eg the fx variable is 2d (lat-lon) and the cube data is 3d (time-lat-lon) - do we not use the fx variable in that case? 2. the fx variable is on a finer or coarser time axis than the cube data - in this case we should do a time regridding operation and still use the fx data. OK these cases are outside the scope of this PR so I'll just keep this in mind for a future feature request :+1: